fuel-vm/src/interpreter/alu/tests.rs

```diff
@@ -0,0 +1,189 @@
+use super::*;
+use fuel_asm::Imm12;
+use std::ops::Div;
+use test_case::test_case;
+
+#[derive(Debug, PartialEq, Eq)]
+struct CommonInput {
+    of: Word,
+    err: Word,
+    pc: Word,
+}
+
+#[test_case(
+    CommonInput { of: 0, err: 0, pc: 0 },
+    0, 0, 0 => Ok((0, CommonInput { of: 0, err: 0, pc: 4 }));
+    "add 0 0"
+)]
+#[test_case(
+    CommonInput { of: 0, err: 0, pc: 0 },
+    0, 1, 1 => Ok((2, CommonInput { of: 0, err: 0, pc: 4 }));
+    "add 1 1"
+)]
+#[test_case(
+    CommonInput { of: 0, err: 0, pc: 0 },
+    0b10, u64::MAX as u128, 1 => Ok((0, CommonInput { of: 1, err: 0, pc: 4 }));
+    "add u64::MAX 1 wrapping"
+)]
+#[test_case(
+    CommonInput { of: 0, err: 0, pc: 0 },
+    0b00, u64::MAX as u128, 1 => Err(RuntimeError::Recoverable(PanicReason::ArithmeticOverflow));
+    "add u64::MAX 1 not wrapping"
+)]
+#[test_case(
+    CommonInput { of: 0, err: 1, pc: 0 },
+    0, 0, 0 => Ok((0, CommonInput { of: 0, err: 0, pc: 4 }));
+    "err is cleared"
+)]
+fn test_add(
+    CommonInput {
+        mut of,
+        mut err,
+        mut pc,
+    }: CommonInput,
+    flag: Word,
+    b: u128,
+    c: u128,
+) -> Result<(Word, CommonInput), RuntimeError> {
+    let common = AluCommonReg {
+        of: RegMut::new(&mut of),
+        err: RegMut::new(&mut err),
+        pc: RegMut::new(&mut pc),
+    };
+    let mut dest = 0;
+    alu_capture_overflow(&mut dest, Reg::new(&flag), common, u128::overflowing_add, b, c)
```

Comment:
I feel like we should consider wrapping add more fully into it's own fn, as this is basically a copy/paste of what add does rather than testing the same exact add logic used by the interpreter..

Reply:
Yeh but this is true for basically all the ops in alu. I guess the point is to more unit test the actual function `alu_capture_overflow`